src/openforms/appointments/api/urls.py

```diff
@@ -1,10 +1,21 @@
 from django.urls import path
 
-from .views import DatesListView, LocationsListView, ProductsListView, TimesListView
+from .views import (
+    CancelAppointmentView,
+    DatesListView,
+    LocationsListView,
+    ProductsListView,
+    TimesListView,
+)
 
 urlpatterns = [
     path("products", ProductsListView.as_view(), name="appointments-products-list"),
     path("locations", LocationsListView.as_view(), name="appointments-locations-list"),
     path("dates", DatesListView.as_view(), name="appointments-dates-list"),
     path("times", TimesListView.as_view(), name="appointments-times-list"),
+    path(
+        "<str:submission_uuid>/cancel",
```

Comment:
```suggestion
        "<uuid:submission_uuid>/cancel",
```
https://docs.djangoproject.com/en/2.2/topics/http/urls/#path-converters there is a `uuid` converter

Reply:
The reason for using `str` is that the UUIDs of the submissions are stored in the session as strings, so `ActiveSubmissionPermission.has_object_permission` does `submission_id in active_submissions` but if `submission_id` at this point is a UUID then this check will fail and the user would get a 403.

I could update https://github.com/open-formulieren/open-forms/blob/master/src/openforms/submissions/api/permissions.py#L36 to be `return str(submission_id) in active_submissions` if you think that's a better solution?  Or maybe there is a better way solution?

Reply:
Yes, go ahead with updating the permission class.

And if we were to keep the `str` pattern, _add a comment explaining the non-obvious choice_. We really can't guess from the code itself why the code was written that way, so document design decisions.